node/external/transactionAPI/refundDetector.go

```diff
@@ -0,0 +1,45 @@
+package transactionAPI
+
+import (
+	"bytes"
+	"encoding/hex"
+
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+type refundDetectorInput struct {
+	Value         string
+	Data          []byte
+	ReturnMessage string
+	GasLimit      uint64
+}
+
+type refundDetector struct {
+}
+
+func newRefundDetector() *refundDetector {
+	return &refundDetector{}
+}
+
+// Also see: https://github.com/ElrondNetwork/elastic-indexer-go/blob/master/process/transactions/scrsDataToTransactions.go
+func (detector *refundDetector) isRefund(result refundDetectorInput) bool {
```

Comment:
maybe  can rename the `result` variable in something more specific like `input` or `scrInput`

Reply:
Renamed to `input`.